src/main/java/br/com/alura/services/Fechamento.java

```diff
@@ -0,0 +1,134 @@
+package br.com.alura.services;
+
+import br.com.alura.comex.CategoriasProcessadas;
+import br.com.alura.comex.Cliente;
+import br.com.alura.comex.ClientesDoSistema;
+import br.com.alura.comex.Pedido;
+import br.com.alura.comex.PedidosDeUmFechamento;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+public class Fechamento {
+
+  CategoriasProcessadas categoriasProcessadas = new CategoriasProcessadas();
+  PedidosDeUmFechamento pedidosDeUmFechamento;
+  ClientesDoSistema clientesDoSistema;
+
+  public Fechamento(
+    PedidosDeUmFechamento pedidosDeUmFechamento,
+    ClientesDoSistema clientesDoSistema
+  ) {
+    this.pedidosDeUmFechamento = pedidosDeUmFechamento;
+    this.clientesDoSistema = clientesDoSistema;
+
+    verificaSeEstaNaCategoria();
+
+    new RelatorioSintetico(this);
+  }
+
+  public Pedido getPedidoMaisBarato() throws IllegalStateException {
+    Optional<Pedido> pedidoMaisBarato =
+      this.pedidosDeUmFechamento.getPedidos()
+        .stream()
+        .min(Comparator.comparing(Pedido::getValorTotal));
+    if (pedidoMaisBarato.isPresent()) return pedidoMaisBarato.get();
+    throw new IllegalStateException("Não existe pedido mais caro");
```

Comment:
Aqui, poderia ser mais fluente!

```suggestion
   return
      this.pedidosDeUmFechamento.getPedidos()
        .stream()
        .min(Comparator.comparing(Pedido::getValorTotal))
        .orElseThrow(() -> new IllegalStateException("Não existe pedido mais caro"));
```